packages/docusaurus/src/client/PendingNavigation.tsx

```diff
@@ -56,56 +52,32 @@ class PendingNavigation extends React.Component<Props, State> {
     // Save the location first.
     this.previousLocation = this.props.location;
     this.setState({nextRouteHasLoaded: false});
-    this.startProgressBar();
+    this.previousRouteUpdateCallback =
+      this.clientLifecyclesDispatcher.current?.onRouteUpdate({
+        previousLocation: this.previousLocation,
+        location: nextLocation,
+      }) || null;
 
     // Load data while the old screen remains.
     preload(nextLocation.pathname)
-      .then(() => {
-        clientLifecyclesDispatcher.onRouteUpdate({
-          previousLocation: this.previousLocation,
-          location: nextLocation,
-        });
-        this.setState({nextRouteHasLoaded: true}, this.stopProgressBar);
-        const {hash} = nextLocation;
-        if (!hash) {
-          window.scrollTo(0, 0);
-        } else {
-          const id = decodeURIComponent(hash.substring(1));
-          const element = document.getElementById(id);
-          element?.scrollIntoView();
-        }
-      })
+      .then(() => this.setState({nextRouteHasLoaded: true}))
       .catch((e) => console.warn(e));
     return false;
   }
 
-  private clearProgressBarTimeout() {
-    if (this.progressBarTimeout) {
-      window.clearTimeout(this.progressBarTimeout);
-      this.progressBarTimeout = null;
-    }
-  }
-
-  private startProgressBar() {
-    this.clearProgressBarTimeout();
-    this.progressBarTimeout = window.setTimeout(() => {
-      clientLifecyclesDispatcher.onRouteUpdateDelayed({
-        location: this.props.location,
-      });
-      nprogress.start();
-    }, this.props.delay);
-  }
-
-  private stopProgressBar() {
-    this.clearProgressBarTimeout();
-    nprogress.done();
-  }
-
   override render(): JSX.Element {
     const {children, location} = this.props;
+    this.previousRouteUpdateCallback?.();
```

Comment:
I don't know where's the best place to place this cleanup. Either in `render`, or in the callback of `setState` (where we previously had `this.stopProgressBar`)

Reply:
Actually... I'll move it to `setState`.